ui/seek_bar.js

```diff
@@ -120,6 +163,97 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
+      const scale = (max - min) / rect.width;
+      // Mouse position in units, which may be outside the allowed range.
+      const value = Math.round(min + scale * mousePosition);
+      // Show Thumbnail
+      this.showThumbnail_(mousePosition, value);
+    });
+
+    this.eventManager.listen(this.bar, 'keydown', (event) => {
+      this.hideThumbnailTimer_.stop();
+      const leftKeyCode = 37;
+      const rightKeyCode = 39;
+      // Left and Right only
+      if (event.keyCode != leftKeyCode && event.keyCode != rightKeyCode) {
+        return;
+      }
+      this.isMoving_ = true;
+      const min = parseFloat(this.bar.min);
+      const max = parseFloat(this.bar.max);
+      if (event.keyCode == leftKeyCode) {
+        this.bar.value = parseFloat(this.bar.value) - (max - min) * 0.025;
+      } else if (event.keyCode == rightKeyCode) {
+        this.bar.value = parseFloat(this.bar.value) + (max - min) * 0.025;
+      }
+      const rect = this.bar.getBoundingClientRect();
+      const value = Math.round(this.bar.value);
+      const scale = (max - min) / rect.width;
+      const position = (value - min) / scale;
+      this.showThumbnail_(position, value);
+      event.preventDefault();
+    });
+
+    this.eventManager.listen(this.bar, 'keyup', () => {
+      if (this.isMoving_) {
+        this.isMoving_ = false;
+        this.hideThumbnailTimer_.stop();
+        this.hideThumbnailTimer_.tickAfter(/* seconds= */ 0.25);
+      }
+    });
+
+    if (navigator.maxTouchPoints > 0) {
+      const touchMove = (event) => {
+        this.isMoving_ = true;
+        const rect = this.bar.getBoundingClientRect();
+        const min = parseFloat(this.bar.min);
+        const max = parseFloat(this.bar.max);
+        // Pixels from the left of the range element
+        const touchPosition = event.changedTouches[0].clientX - rect.left;
+        // Pixels per unit value of the range element.
+        const scale = (max - min) / rect.width;
+        // Mouse position in units, which may be outside the allowed range.
+        const value = Math.round(min + scale * touchPosition);
+        // Show Thumbnail
+        this.showThumbnail_(touchPosition, value);
+        // Update the bar
+        this.bar.value = value;
+        event.preventDefault();
+      };
+      const touchEnd = () => {
+        if (this.isMoving_) {
+          this.isMoving_ = false;
+          this.hideThumbnail_();
+        }
+      };
+      this.eventManager.listen(this.bar, 'touchstart', touchMove);
+      this.eventManager.listen(this.bar, 'touchmove', touchMove);
+      this.eventManager.listen(this.bar, 'touchend', touchEnd);
+      this.eventManager.listen(this.bar, 'touchcancel', touchEnd);
+    }
+
+    this.eventManager.listen(this.bar, 'blur', () => {
+      if (this.isMoving_) {
+        this.isMoving_ = false;
+        this.hideThumbnail_();
+      }
+    });
+
+    this.eventManager.listen(this.container, 'mouseleave', () => {
+      this.hideThumbnail_();
```

Comment:
This technically causes the thumbnail to flicker if you move your mouse off the bar while moving the bar with keyboard controls, but it's such a subtle effect that it's probably not worth fixing.

Reply:
If I change it to `this.hideThumbnailTimer_.tickAfter(/* seconds= */ 0.25);`, do you think it will have a better effect?

Reply:
Done!